### ISSUE TITLE
CommitTransaction: cannot commit - no transaction is active

### ISSUE THREAD

**Issue by muuvmuuv** [capacitor-community/sqlite#414]
**Describe the bug**

I am doing a simple db.execute `DELETE FROM table;` and got this since a few release (cannot tell since when, sry). Any idea what could cause this? I do a `PRAGMA foreign_keys = OFF;` before (to easier wipe all data in that case). All other SQL execute just fine before.

```
Execute Error Error: Execute: ExecuteSQL: ExecuteSQL: CommitTransaction: cannot commit - no transaction is active : RollbackTransaction: cannot rollback - no transaction is active
```

**To Reproduce**

1. Simple delete from

**Expected behavior**
More detailed errors...

**Screenshots**

**Desktop (please complete the following information):**

- OS: macOS
- Browser: Brave
- Version: latest

**Smartphone (please complete the following information):**

**Additional context**

**Comment by jepiqueau** [comment]
@muuvmuuv the PRAGMA foreign_keys is manager directly by the plugin. Can you share a simple app demonstrating the issue

**Comment by muuvmuuv** [comment]
What do you mean, I don't need to do this manually while using execute? Isn't this a bit antipattern to be on by default?

**Comment by jepiqueau** [comment]
@muuvmuuv Is it in the iOS part of the plugin that you get the error ? Did you set-up a sync table ? Do you have a `sql_deleted` field in your table? It would be easier for me to progress if i got a shred example.

**Comment by muuvmuuv** [comment]
Browser, didn't test on iOS yet, had no time to dig into it further...
We don't have either a sync table or sql_deleted field.

**Comment by jepiqueau** [comment]
@muuvmuuv are you using typeorm?

**Comment by muuvmuuv** [comment]
We aren't using any ORM.

**Comment by jepiqueau** [comment]
@muuvmuuv Are you also use autosave? look at [typeorm issue#9100](https://github.com/typeorm/typeorm/issues/9100). 
Where you create the jeep-sqlite element can you try with autosave = false  and do a call to autoSave method only when a certain number of transaction are done and tell me if it works
ie
```
    const jeepEl = document.createElement("jeep-sqlite");
    document.body.appendChild(jeepEl);
    jeepEl.autoSave = false;

```

**Comment by jepiqueau** [comment]
@muuvmuuv The autosave = true each time there is a transaction which is committed an export of the database is done to localforage to store in the jeepSqliteStore. If there is a new transaction started during this process there is a chance that you get that message depending also of the size of the database

**Comment by jepiqueau** [comment]
@muuvmuuv Are you sure that you do not use BEGIN TRANSACTION, COMMIT TRANSACTION or ROLLBACK TRANSACTION   in a sql statement?

**Comment by jepiqueau** [comment]
@muuvmuuv are you using `jeep-sqlite` alone or through the @capacitor-community/sqlite

**Comment by jepiqueau** [comment]
@muuvmuuv if it is through @capacitor-community/sqlite can you show me the code where you instantiate 'jeep-sqlite'

**Comment by muuvmuuv** [comment]
Through `@capacitor-community/sqlite`:

```ts
  async init() {
    try {
      if (isWeb) {
        await customElements.whenDefined("jeep-sqlite")
        const jeepSqliteElement = document.querySelector("jeep-sqlite")
        if (jeepSqliteElement != null) {
          await this.sqlite.initWebStore()
        }
      }

      // Migrate cordova database
      await this.migrateCordovaDatabase()

      // Make sure JavaScript and native connections are in sync
      const connection = await this.sqlite.checkConnectionsConsistency()
      this.log.debug(LogStack.DatabaseService, "Connection consistent", connection.result)

      // Connect to database
      const connected = await this.sqlite.isConnection(this.DB_NAME, false)
      if (connected.result) {
        this.log.info(LogStack.DatabaseService, "Retrieving existing connection...")
        this.#database = await this.sqlite.retrieveConnection(this.DB_NAME, false)
      } else {
        this.log.info(LogStack.DatabaseService, "Creating new DB connection...")
        this.#database = await this.sqlite.createConnection(
          this.DB_NAME,
          false,
          "no-encryption",
          this.DB_VERSION,
          false,
        )
      }
      await this.#database.open()

      // Run echo test
      await this.sqlite.echo("entergon")

      // Execute revisions
      const executedRevisions = await this.getRevisions()
      const revisionFactory = []
      this.environmentInjector.runInContext(() => {
        for (const Revision of revisions) revisionFactory.push(new Revision(executedRevisions))
      })
      for (const revision of revisionFactory) {
        await revision.run()
      }
    } catch (error) {
      this.log.error(LogStack.DatabaseService, `Fatal`, error)
      throw error
    }
  }
```

**Comment by muuvmuuv** [comment]
We don't use TRANSACTION but `executeSet` instead. And we don't call any `TRANSACTION` calls manually.

**Comment by muuvmuuv** [comment]
So, I think I got it, it must be something with JavaScript runtime, these `(Revision)` run in a loop. Each screenshot is a reload. So its executing the SQL for one revision class and fails on the second loop revision entry. Reload will continue where it failed and failed right after...

![SCR-20230530-pawd](https://github.com/capacitor-community/sqlite/assets/11534760/626bbd7e-d200-41ea-ae95-ac418a998248)
![SCR-20230530-payb](https://github.com/capacitor-community/sqlite/assets/11534760/cff51da7-4773-415c-95ca-730a6087098c)
![SCR-20230530-payz](https://github.com/capacitor-community/sqlite/assets/11534760/076c00cc-9a93-45d1-80f3-e8eec651df47)

**Comment by muuvmuuv** [comment]
Setting transaction to false on `execute` not `executeSet` solved it. Thought that its "false" by default since it accepts not set and undefined?

**Comment by muuvmuuv** [comment]
Yeah, followed that at the start but our database service has changed a lot since the start of switching to your lib, so might not seen that explicit parameter declaration

**Comment by jepiqueau** [comment]
@muuvmuuv the idea of setting transaction to false on any execute, executeSet and run was made for the use with typeOrm. 
it also can be use to able a developer to manage itself the transaction through the BEGIN TRANSACTION, COMMIT TRANSACTION & ROLLBACK TRANSACTION through execute method

**Comment by jepiqueau** [comment]
@muuvmuuv take care that if you set it to false the execute is made obviously without Transaction process

**Comment by muuvmuuv** [comment]
Yeah, will keep that in mind, but until now that is fine, we will switch to an ORM in a future version of our app but now its only legacy code so no need to put that energy in.

**Comment by jepiqueau** [comment]
@muuvmuuv  Ok you are welcome
